tools/encyclopedia/encyclopedia_fasta_to_prosit_csv.xml

```diff
@@ -0,0 +1,61 @@
+<tool id="encyclopedia_fasta_to_prosit_csv" name="ConvertFastaToPrositCSV" version="@VERSION@.0" python_template_version="3.5">
+    <description>Create Prosit CSV Input From a Protein FASTA</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="aggressive"><![CDATA[
+        EncyclopeDIA -convert -fastatoprositcsv 
+        -i '$i'
+        -minCharge $minCharge
+        -maxCharge $maxCharge
+        -defaultCharge $defaultCharge
+        -maxMissedCleavage $maxMissedCleavage
+        -minMz $minMz
+        -maxMz $maxMz
+        -defaultNCE $defaultNCE
+        #if $enzyme
+            -enzyme '$enzyme'
+        #end if
+        -o '$output'
+    ]]></command>
+    <inputs>
+        <param argument="-i" type="data" format="fasta" label="Protein Fasta database"/>
+	<param argument="-minCharge" type="integer" value="2" min="1" max="6" label="minCharge" help="(default: 2)"/>
+	<param argument="-maxCharge" type="integer" value="3" min="1" max="6" label="maxCharge" help="(default: 3)"/>
+	<param argument="-defaultCharge" type="integer" value="3" min="1" max="4" label="defaultCharge" help="(default: 3)"/>
+	<param argument="-maxMissedCleavage" type="integer" value="1" min="0" max="3" label="maxMissedCleavage" help="(default: 1)"/>
+	<param argument="-minMz" type="float" value="396.4" min="150.0" max="1600.0" label="minMz" help="(default: 396.4)"/>
+	<param argument="-maxMz" type="float" value="1002.7" min="150.0" max="1600.0" label="maxMz" help="(default: 1002.7)"/>
+	<param argument="-defaultNCE" type="integer" value="33" min="25" max="40" label="Normalized Collision Energy" help="(default: 33)"/>
+        <param argument="-enzyme" type="select" optional="true" label="Digestion Enzyme" help="(default: Trypsin)">
+            <option value="Trypsin">Trypsin</option>
+            <option value="Lys-C">Lys-C</option>
+            <option value="Lys-N">Lys-N</option>
+            <option value="Arg-C">Arg-C</option>
+            <option value="Glu-C">Glu-C</option>
+            <option value="CNBr">CNBr</option>
+            <option value="Chymotrypsin">Chymotrypsin</option>
+            <option value="Pepsin A">Pepsin A</option>
+            <option value="No Enzyme">No Enzyme</option>
+        </param>
+    </inputs>
+    <outputs>
+        <data name="output" format="csv" label="${i.name} Prosit_input.csv"/>
```

Comment:
So far, prosit requires the CSV format as input.  https://github.com/kusterlab/prosit
We hope to make a galaxy tool prosit.  This would be the workflow:
fasta -> **encyclopedia_fasta_to_prosit_csv** -> CSV -> **prosit** -> CSV -> **encyclopedia_prosit_csv_to_library** -> dlib

Reply:
how likely is it that those CSVs in the middle needs to be manipulated by the user, e.g. filtered ...